File: RectangleDemoCs/Rectangle.cs
using System;
using RectangleDemoCs.Interfaces;

namespace RectangleDemoCs
{
    public class Rectangle : IShape
    {
        public decimal Length { get; private set; }
        public decimal Width { get; private set; }

        public Rectangle(decimal length, decimal width)
        {
            if (length <= 0) throw new ArgumentOutOfRangeException("length", "length must be positve");
            if (width <= 0) throw new ArgumentOutOfRangeException("width", "width must be positve");
            
            Length = length;
            Width = width;
        }

        public decimal Area
        {
            get { return Length * Width; }
        }

        public decimal Perimeter
        {
            get { return (Length * 2) + (Width * 2); }
        }

        public bool IsSquare()
        {
            return Length == Width;
        }
    }
}

File: RectangleDemoFs/Rectangle.fs
namespace RectangleDemoFs

open System

type Rectangle(length: decimal, width: decimal) =
    do
        if length <= 0m then raise (ArgumentOutOfRangeException("length"))
        if width <= 0m then raise (ArgumentOutOfRangeException("width"))


    member this.IsSquare() =
        length.Equals width

    member val Length =
        length

    member val Width =
        width

    member val Area =
        length * width

    member val Perimeter =
        (length * 2m) + (width * 2m)

    